lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1823,9 +1823,11 @@ static bool ParseSILArgs(SILOptions &Opts, ArgList &Args,
   Opts.EnableStackProtection =
       Args.hasFlag(OPT_enable_stack_protector, OPT_disable_stack_protector,
                    Opts.EnableStackProtection);
-  Opts.EnableMoveInoutStackProtection =
-      Args.hasFlag(OPT_enable_move_inout_stack_protector, OPT_disable_stack_protector,
-                   Opts.EnableMoveInoutStackProtection);
+  Opts.EnableMoveInoutStackProtection = Args.hasArg(
+      OPT_enable_move_inout_stack_protector, OPT_disable_stack_protector,
+      Opts.EnableMoveInoutStackProtection);
+  Opts.EnableImportPtrauthFieldFunctionPointers =
+      Args.hasArg(OPT_enable_import_ptrauth_field_function_pointers);
```

Comment:
Nitpick for a follow-up .

I would recommend adding an option to disable the feature `OPT_disable_import_ptrauth_field_function_pointers`. And the using the pattern:

```
Opts.EnableImportPtrauthFieldFunctionPointers = Args.hasArg(OPT_enable_import_ptrauth_field_function_pointers,
                      OPT_disable_import_ptrauth_field_function_pointers,
                       Opts.EnableImportPtrauthFieldFunctionPointers );
```

This will enable/disable the feature if one of the flags is passed to the invocation and use the default from the initialized `EnableImportPtrauthFieldFunctionPointers` field if no flag is used.